pypeit/reduce.py

```diff
@@ -1195,128 +1198,188 @@ def find_objects_pypeline(self, image, std_trace=None,
+        for parameter definitions.
+        """
+        msgs.info("Performing joint global sky subtraction")
+        # Mask objects using the skymask? If skymask has been set by objfinding, and masking is requested, then do so
+        nslits = self.slits.spat_id.size
+        skymask_now = skymask if (skymask is not None) else np.ones_like(self.sciImg.image, dtype=bool)
         self.global_sky = np.zeros_like(self.sciImg.image)
+        thismask = (self.slitmask > 0)
+        inmask = ((self.sciImg.fullmask == 0) & thismask & skymask_now).astype(np.bool)
+        # Convert the wavelength image to A/pixel, registered at pixel 0 (this gives something like
+        # the tilts frame, but conserves wavelength position in each slit)
+        wavemin = self.waveimg[self.waveimg != 0.0].min()
+        tilt_wave = (self.waveimg - wavemin) / (self.waveimg.max() - wavemin)
+
         # Parameters for a standard star
+        sigrej = 3.0
         if self.std_redux:
             sigrej = 7.0
             update_crmask = False
             if not self.par['reduce']['skysub']['global_sky_std']:
                 msgs.info('Skipping global sky-subtraction for standard star.')
                 return self.global_sky
-        else:
-            sigrej = 3.0
 
-        gdslits = np.where(np.logical_not(self.reduce_bpm))[0]
-
-        # Mask objects using the skymask? If skymask has been set by objfinding, and masking is requested, then do so
-        skymask_now = skymask if (skymask is not None) else np.ones_like(self.sciImg.image, dtype=bool)
-
-        if self.par['reduce']['skysub']['joint_fit']:
-            msgs.info("Performing joint global sky subtraction")
-            thismask = (self.slitmask > 0)
-            inmask = ((self.sciImg.fullmask == 0) & thismask & skymask_now).astype(np.bool)
-            # Convert the wavelength image to A/pixel, registered at pixel 0 (this gives something like
-            # the tilts frame, but conserves wavelength position in each slit)
-            tilt_wave = (self.waveimg - self.waveimg.min()) / (self.waveimg.max() - self.waveimg.min())
-            # Find sky
+        # Iterate to use a model variance image
+        numiter = 4
+        model_ivar = self.sciImg.ivar.copy()
+        for nn in range(numiter):
+            msgs.info("Performing iterative joint sky subtraction - ITERATION {0:d}/{1:d}".format(nn+1, numiter))
             self.global_sky[thismask] \
-                = skysub.global_skysub(self.sciImg.image, self.sciImg.ivar, tilt_wave,
+                = skysub.global_skysub(self.sciImg.image, model_ivar, tilt_wave,
                                        thismask, self.slits_left, self.slits_right, inmask=inmask,
                                        sigrej=sigrej, trim_edg=trim_edg,
-                                       bsp=self.par['reduce']['skysub']['bspline_spacing'],
+                                       bsp=self.par['reduce']['skysub']['bspline_spacing'] / np.sqrt(nslits),
```

Comment:
I think you are adding this factor here because you are better sampled given the number of slits. I'm not sure this is the correct thing to do and I worry it results in overfitting in addition to slowing things down. Even for very long slits, we usually set the bspline spacing to like 0.7 pixels or so. Unless you are super agressively spectrally sampled, I'm not sure this is doing anything, and it can lead to artifacts. Anyway, you can easily inspect the fits and see if this helps with the debug option.